components/src/main/java/dev/arkbuilders/components/filepicker/ArkFilePickerFragment.kt

```diff
@@ -96,7 +97,24 @@ open class ArkFilePickerFragment :
 
     private val mPinFileCallback = object : PinFileCallback {
         override fun onPinFileClick(file: Path) {
-            pinFile(file)
+            viewModel.pinFile(file, mCurrentRootsWithFavorites)
+        }
+    }
```

Comment:
You don't need this callback. 
Just paste `viewModel.pinFile(file, mCurrentRootsWithFavorites)`  where it is called

Also you don't need `mCurrentRootsWithFavorites`, it’s already in the viewmodel state. To get it inside the viewmodel use:
```
intent {
    state.rootsWithFavs
}
```